### PR TITLE
Initial commit for 0.12.0 release

### PR DESCRIPTION
**Pull request by IsraelMiles**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your PR is merged into master branch, a new PR will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repo and upgrade the version in the PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
quantum

**Review by kairu-ms — APPROVED** [review]

File: src/quantum/HISTORY.rst [hunk 1]
```diff
@@ -3,6 +3,12 @@
 Release History
 ===============
 
+0.12.0
+++++++
+* [2022-01-25] Version intended to work with QDK version v0.22.186636
+* Updated data plane generated client to API version 2021-11-01-preview.
+* Added support for displaying price estimate for a job when available.
```
Comment (ricardo-espinoza) [inline src/quantum/HISTORY.rst:10]:
```suggestion
* Added support for displaying cost estimate for a job when available.
```

Comment (ricardo-espinoza) [reply src/quantum/HISTORY.rst:10]: Per discussion in email, let's call this as "cost estimate instead"

**Review by ricardo-espinoza — CHANGES_REQUESTED** [review]

File: src/quantum/HISTORY.rst (hunk 1 above)
Comment (ricardo-espinoza) [reply src/quantum/HISTORY.rst:10]: Also, please update the string in [this line](https://github.com/IsraelMiles/azure-cli-extensions/blob/israelmiles/release/0.12.0/src/quantum/azext_quantum/commands.py#L46) to "Cost estimate", as below:

            `transformed_result['Cost estimate'] = price`

**Review by ricardo-espinoza — APPROVED** [review]

File: src/quantum/HISTORY.rst [hunk 2]
```diff
@@ -3,6 +3,12 @@
 Release History
 ===============
 
+0.12.0
+++++++
+* [2022-01-25] Version intended to work with QDK version v0.22.186636
```
Comment (anjbur) [inline src/quantum/HISTORY.rst:8]:
```suggestion
* [2022-01-25] Version intended to work with QDK version v0.22.187631
```

**Pull request merged**